composer/callbacks/checkpoint_saver.py

```diff
@@ -320,6 +383,43 @@ def __init__(
 
         self.start_batch = None
 
+        self.remote_uploader = None
+        backend, _, _ = parse_uri(save_folder)
+        self.symlink_file_tasks: list[tuple[str, str]] = []
+        self.rank_saves_remote_symlinks: bool = False
+        self.tmp_dir_for_symlink = tempfile.TemporaryDirectory()
+        self.num_concurrent_uploads = num_concurrent_uploads
+        self.symlink_upload_executor = None
+        self.is_remote_upload_failed = None
+        self.symlink_upload_futures = []
+        self.upload_timeout_in_seconds = upload_timeout_in_seconds
+        # Allow unit test to override this to make it faster
+        self._symlink_upload_wait_before_next_try_in_seconds = 30.0
+        self.pid = os.getpid()
+        self.symlink_count = 0
+
+        if backend != '':
+            if backend == 'wandb':
+                raise NotImplementedError(
+                    f'There is no implementation for WandB via URI. Please use '
+                    'WandBLogger with log_artifacts set to True.',
+                )
+            elif backend not in ['s3', 'oci', 'gs', 'azure', 'dbfs']:
+                raise NotImplementedError(
+                    f'There is no implementation for the cloud backend {backend} via URI. Please use '
+                    'one of the supported object stores.',
```

Comment:
Can you add `['s3', 'oci', 'gs', 'azure', 'dbfs']` to the error message for clarity to the user?